### PR TITLE
feat: add logic to handle node public key submissions only starting from the current key rotation counter

### PR DESCRIPTION
**Pull request by jorgeantonio21**

### CODE REVIEW

File: atoma-bin/atoma_node.rs [hunk 1]
```diff
@@ -225,16 +226,17 @@ async fn main() -> Result<()> {
         register_on_proxy(&config.proxy, *node_small_id, &keystore, args.address_index).await?;
     }
 
-    let (compute_shared_secret_sender, compute_shared_secret_receiver) =
+    let (compute_shared_secret_sender, _compute_shared_secret_receiver) =
         tokio::sync::mpsc::unbounded_channel();
 
+    #[cfg(feature = "tdx")]
     spawn_with_shutdown(
```
Comment (Cifko) [inline atoma-bin/atoma_node.rs:233]: you should keep the `handle` that's returned from the spawn_with_shutdown and put that into the `try_join` at L380.

File: atoma-bin/atoma_node.rs [hunk 2]
```diff
@@ -225,16 +226,17 @@ async fn main() -> Result<()> {
         register_on_proxy(&config.proxy, *node_small_id, &keystore, args.address_index).await?;
     }
 
-    let (compute_shared_secret_sender, compute_shared_secret_receiver) =
+    let (compute_shared_secret_sender, _compute_shared_secret_receiver) =
         tokio::sync::mpsc::unbounded_channel();
 
+    #[cfg(feature = "tdx")]
```
Comment (Cifko) [inline atoma-bin/atoma_node.rs:232]: if we want to hide this behind a feature flag, maybe we can put all TDX stuff behind a feature flag, that will reduce the size/compilation time when building without it. But I'm not sure how the code acts when it doesn't have confidentiality support but it receives confidential compute request anyway, what happens than? Maybe we can put everything behind feature flag and just don't expose the endpoint at all. Any idea what's better?

**Review by Cifko — APPROVED** [review]
LGTM just add the handle the other comment doesn't need to resolve into complex more PR ;-)

File: atoma-bin/atoma_node.rs (hunk 1 above)
Comment (jorgeantonio21) [reply atoma-bin/atoma_node.rs]: I see, applied changes.

**Review by Cifko — APPROVED** [review]

**Comment by jorgeantonio21** [comment]
Addresses #291

**Pull request merged**